hal/network/ncp_client/quectel/quectel_ncp_client.cpp

```diff
@@ -2000,8 +2016,16 @@ int QuectelNcpClient::modemInit() const {
 
     // DTR=0: normal mode, DTR=1: sleep mode
     // NOTE: The BGDTR pins is inverted
-    conf.value = 1;
-    CHECK(hal_gpio_configure(BGDTR, &conf, nullptr));
+    if (BGDTR != PIN_INVALID) {
+        conf.value = 1;
+        CHECK(hal_gpio_configure(BGDTR, &conf, nullptr));
+    }
+
+#if HAL_PLATFORM_CELLULAR_MODEM_VOLTAGE_TRANSLATOR
```

Comment:
How does this work for other platforms such as M SoM? `BUFEN` is not defined there.

Reply:
I flipped the default for `HAL_PLATFORM_CELLULAR_MODEM_VINT_INVERTED` to be 1, and `HAL_PLATFORM_CELLULAR_MODEM_VOLTAGE_TRANSLATOR` to be 1, and made sure `electron/hal_platform_config.h` contains overrides to 0.

Reply:
@technobly This is still broken:
```
  /home/avtolstoy/dev/device-os-2/modules/msom/system-part1/makefile /home/avtolstoy/dev/device-os-2/modules/msom/user-part/makefile
In file included from ./src/rtl872x/gsm0710muxer/platform.h:24,
                 from ../gsm0710muxer/gsm0710muxer/include/gsm0710muxer/muxer.h:25,
                 from ./src/rtl872x/gsm0710muxer/channel_stream.h:21,
                 from network/ncp_client/quectel/quectel_ncp_client.h:28,
                 from network/ncp_client/quectel/quectel_ncp_client.cpp:22:
network/ncp_client/quectel/quectel_ncp_client.cpp: In member function 'int particle::QuectelNcpClient::modemInit() const':
network/ncp_client/quectel/quectel_ncp_client.cpp:2027:30: error: 'BUFEN' was not declared in this scope
 2027 |     CHECK(hal_gpio_configure(BUFEN, &conf, nullptr));
```

None of the existing platforms that use Quectel NCP client have `BUFEN`. IMO `HAL_PLATFORM_CELLULAR_MODEM_VOLTAGE_TRANSLATOR` needs to be set by default to `0` to simplify things, and set to 1 for Boron, B SoM (both SARA) and Electron 2 (Quectel).